cpu/sam0_common/periph/rtc_rtt.c

```diff
@@ -337,31 +337,36 @@ int rtc_tamper_register(gpio_t pin, gpio_flank_t flank)
     return 0;
 }
 
-static void _unlock(void *m)
-{
-    mutex_unlock(m);
-}
-
 void rtc_tamper_enable(void)
 {
-    mutex_t m = MUTEX_INIT;
+    DEBUG("enable tamper\n");
 
     /* clear tamper id */
     RTC->MODE0.TAMPID.reg = 0xF;
 
     /* work around errata 2.17.4:
      * ignore the first tamper event on the rising edge */
     if (RTC->MODE0.TAMPCTRL.reg & RTC_TAMPCTRL_TAMLVL_Msk) {
-        mutex_lock(&m);
-        tamper_cb.cb  = _unlock;
-        tamper_cb.arg = &m;
-    }
 
-    /* enable tamper detect as wake-up source */
-    RTC->MODE0.INTENSET.bit.TAMPER = 1;
+        /* If an RTC alarm happened before, the spurious tamper
+         * event is sometimes generated.
+         * Tamper event must happen within one RTC clock period. */
+        unsigned timeout = CLOCK_CORECLOCK / 32768;
```

Comment:
We could further divide the timeout value here as the `!RTC->MODE0.INTFLAG.bit.TAMPER && --timeout` check takes more than once cycle.

Reply:
Better safe than sorry.